data/LocalStore.js

```diff
@@ -128,11 +143,18 @@ export class LocalStore extends BaseStore {
     // Private Implementation
     //------------------------
     createRecords(rawData, parent = null) {
+        const {idSpec} = this;
+        const idGen = isString(idSpec) ? r => r[idSpec] : idSpec;
+
         return rawData.map(raw => {
             if (this.processRawData) this.processRawData(raw);
 
-            // All records must have a unique, non-null ID - install a generated one if required.
-            if (isNil(raw.id)) raw.id = XH.genId();
+            raw.id = idGen(raw);
+            throwIf(
+                isUndefined(raw.id),
+                'Cannot load record without a unique id.  Provide a unique id on each raw record using the ' +
```

Comment:
This message may be a bit misleading, as we aren't really checking that the id is unique. Maybe something like "Cannot load record without an id. Please ensure the `idSpec` property of this store resolves to a unique identifier" or something?

Reply:
Also curious why we are just checking `isUndefined` here now instead of `isNil`? Are we ok with a record in the store having a `null` id?

Reply:
great catch with isUndefined -- I will change that.    Regarding the message, yes was trying to reinforce to the user that when they do specify an id, they ought to make the id unique.  So I guess I can either change the message or do the unique check.

Thanks for the careful look